designs/cache_options.md

```diff
@@ -0,0 +1,170 @@
+Cache Options
+===================
+
+This document describes how we imagine the cache options to look like in
+the future.
+
+## Goals
+
+* Align everyone on what settings on the cache we want to support and
+  their configuration surface
+* Ensure that we support both complicated cache setups and provide an
+  intuitive configuration UX
+
+## Non-Goals
+
+* Line out how the actual implementation of the cache itself will look like.
+  The assumption is that the most granular level we will end up with is
+  "per object multiple namespaces with distinct selectors" and that this
+  can be implemented using a "meta cache" that delegates per object and by
+  extending the current multi-namespace cache
+* Outline any kind of timeline for when these settings will be implemented.
+  Implementation will happen gradually over time whenever someone steps up
+  to do the actual work
+
+## Proposal
+
+
+```
+type CacheSetting struct {
+  // LabelSelector specifies a label selector. A nil value allows to
+  // default this.
+  LabelSelector labels.Selector
+
+  // FieldSelector specifics a field selector. A nil value allows to
+  // default this.
+  FieldSelector fields.Selector
+
+  // Transform specifies a transform func. A nil value allows to default
+  // this.
+  Transform     toolscache.TransformFunc
+
+  // UnsafeDisableDeepCopy specifies if List and Get requests against the
+  // cache should not DeepCopy. A nil value allows to default this.
+  UnsafeDisableDeepCopy *bool
```

Comment:
The double negative here is confusing to read. It's not safe, and it disables behavior, without committing to what the default value would be.

Alternatives to consider:
- _CopyDeeply_
- _CopyDefensively_
- _CopyOnRead_
- _ReturnDefensiveCopies_

Reply:
You want that to be in the field name, and not just in the field's accompanying documentation?